saleor/graphql/menu/mutations.py

```diff
@@ -42,6 +52,45 @@ class Meta:
     def user_is_allowed(cls, user, input):
         return user.has_perm('menu.manage_menus')
 
+    @classmethod
+    def clean_input(cls, info, instance, input, errors):
+        cleaned_input = super().clean_input(info, instance, input, errors)
+        items = []
+        for item in cleaned_input.get('items', []):
+            category = item.get('category')
+            collection = item.get('collection')
+            page = item.get('page')
+            url = item.get('url')
+            if len([i for i in [category, collection, page, url] if i]) > 1:
+                cls.add_error(
+                    errors, 'items', 'More than one item provided.')
+            else:
+                if category:
+                    category = cls.get_node_or_error(
+                        info, category, errors, 'items', Category)
```

Comment:
Small nit here, I'd like to be consistent and use `only_type` keyword as in calls below.